processor/transformprocessor/internal/traces/traces.go

```diff
@@ -0,0 +1,368 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+// pathGetSetter is a getSetter which has been resolved using a path expression provided by a user.
+type pathGetSetter struct {
+	getter exprFunc
+	setter func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{})
+}
+
+func (path pathGetSetter) get(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) interface{} {
+	return path.getter(span, il, resource)
+}
+
+func (path pathGetSetter) set(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{}) {
+	path.setter(span, il, resource, val)
+}
+
+func newGetSetter(val common.Value) (getSetter, error) {
+	if val.Path == nil {
+		return nil, fmt.Errorf("must be a trace path expression")
+	}
+
+	return newPathGetSetter(val.Path.Fields)
+}
+
+func newPathGetSetter(path []common.Field) (getSetter, error) {
+	var getter exprFunc
+	var setter func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{})
+	switch path[0].Name {
+	case "resource":
```

Comment:
It feels like we can only access the resource attributes, what about the dropped atttributes https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/resource/v1/resource.proto#L33 (we need to add this to pdata, will file a bug) or the schema_url in the `ResourceSpan`?